cmake/BLTMacros.cmake

```diff
@@ -1274,96 +1274,92 @@ endmacro(blt_combine_static_libraries)
+    set(singleValuedArgs TARGET CHILDREN PROPERTY_NAME_REGEX PROPERTY_VALUE_REGEX)
     set(multiValuedArgs)
 
-    ## parse the arguments to the macro
+    # parse the arguments to the macro
     cmake_parse_arguments(arg
          "${options}" "${singleValuedArgs}" "${multiValuedArgs}" ${ARGN})
 
-    ## check for required arguments
+    # check for required arguments
     if(NOT DEFINED arg_TARGET)
         message(FATAL_ERROR "TARGET is a required parameter for the blt_print_target_properties macro")
     endif()
 
-    ## check if this is a valid cmake target of blt_registered target
+    # set default values
+    if(NOT DEFINED arg_CHILDREN)
+        set(arg_CHILDREN FALSE)
+    endif()
+
+    if(NOT DEFINED arg_PROPERTY_NAME_REGEX)
+        set(arg_PROPERTY_NAME_REGEX ".*")
+    endif()
+
+    if(NOT DEFINED arg_PROPERTY_VALUE_REGEX)
+        set(arg_PROPERTY_VALUE_REGEX ".*")
+    endif()
+
+    # check if this is a valid cmake target or blt_registered target
     set(_is_cmake_target FALSE)
     if(TARGET ${arg_TARGET})
         set(_is_cmake_target TRUE)
-        message (STATUS "[${arg_TARGET} property] '${arg_TARGET}' is a cmake target")
     endif()
 
     set(_is_blt_registered_target FALSE)
     string(TOUPPER ${arg_TARGET} _target_upper)
     if(_BLT_${_target_upper}_IS_REGISTERED_LIBRARY)
         set(_is_blt_registered_target TRUE)
-        message (STATUS "[${arg_TARGET} property] '${arg_TARGET}' is a blt_registered target")
     endif()
 
     if(NOT _is_cmake_target AND NOT _is_blt_registered_target)
         message (STATUS "[blt_print_target_properties] Invalid argument '${arg_TARGET}'. "
                          "This macro applies only to valid cmake targets or blt_registered targets.")
-    endif()
-
-
-    if(_is_cmake_target)
-        ## Solution adapted from https://stackoverflow.com/q/32183975
-        ## Create list of cmake properties
-        set(_property_list)
-        execute_process(COMMAND cmake --help-property-list OUTPUT_VARIABLE _property_list)
-        string(REGEX REPLACE ";" "\\\\;" _property_list "${_property_list}")
-        string(REGEX REPLACE "\n" ";" _property_list "${_property_list}")
-        blt_filter_list(TO _property_list REGEX "^LOCATION$|^LOCATION_|_LOCATION$" OPERATION "exclude")
-        blt_list_remove_duplicates(TO _property_list)   
-
-        ## For interface targets, filter against whitelist of valid properties
-        get_property(_target_type TARGET ${arg_TARGET} PROPERTY TYPE)
-        if("${_target_type}" STREQUAL "INTERFACE_LIBRARY")
-            blt_filter_list(TO _property_list
-                            REGEX "^(INTERFACE_|IMPORTED_LIBNAME_|COMPATIBLE_INTERFACE_|MAP_IMPORTED_CONFIG_)|^(NAME|TYPE|EXPORT_NAME)$"
-                            OPERATION "include")
+    else()
+        # print properties
+        blt_print_target_properties_private(TARGET ${arg_TARGET}
+                                            PROPERTY_NAME_REGEX ${arg_PROPERTY_NAME_REGEX})
```

Comment:
Note: The macro doesn't appear to use `PROPERTY_VALUE_REGEX`.

I'm guess this will be added in the future?

Reply:
yes, this was a mistake. this mistake actually revealed more mistakes in `blt_print_target_properties_private`:

- my regex `MATCHES` was checking `_propval` before `get_target_property` ran in the cmake target section
- and i had`_propval` in the blt registered target section (when it should be `${${prop}}`)

i've fixed these and they'll be up on github soon.